src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapGet_NoParam_StringReturn_WithFilter.generated.txt

```diff
@@ -0,0 +1,206 @@
+
+    %GENERATEDCODEATTRIBUTE%
+    file static class GeneratedRouteBuilderExtensionsCore
+    {
+        internal static class GenericThunks<T>
+        {
+            public static readonly System.Collections.Generic.Dictionary<(string, int), (MetadataPopulator, RequestDelegateFactoryFunc)> map = new()
+            {
+
+            };
+        }
+
+        internal static readonly System.Collections.Generic.Dictionary<(string, int), (MetadataPopulator, RequestDelegateFactoryFunc)> map = new()
+        {
+                        [(@"TestMapActions.cs", 15)] = (
+           (methodInfo, options) =>
+            {
+                if (options == null)
+                {
+                    return new RequestDelegateMetadataResult { EndpointMetadata = ReadOnlyCollection<object>.Empty };
+                }
+                options.EndpointBuilder.Metadata.Add(new SourceKey(@"TestMapActions.cs", 15));
+                return new RequestDelegateMetadataResult { EndpointMetadata = options.EndpointBuilder.Metadata.AsReadOnly() };
+            },
+            (del, options, inferredMetadataResult) =>
+            {
+                var handler = (System.Func<string>)del;
+                EndpointFilterDelegate? filteredInvocation = null;
+
+                if (options.EndpointBuilder.FilterFactories.Count > 0)
+                {
+                    filteredInvocation = GeneratedRouteBuilderExtensionsCore.BuildFilterDelegate(ic =>
+                    {
+                        if (ic.HttpContext.Response.StatusCode == 400)
+                        {
+                            return System.Threading.Tasks.ValueTask.FromResult<object?>(Results.Empty);
+                        }
+                        return System.Threading.Tasks.ValueTask.FromResult<object?>(handler());
+                    },
+                    options.EndpointBuilder,
+                    handler.Method);
+                }
+
+                System.Threading.Tasks.Task RequestHandler(Microsoft.AspNetCore.Http.HttpContext httpContext)
+                {
+                        var result = handler();
+                        return httpContext.Response.WriteAsync(result);
+                }
+                async System.Threading.Tasks.Task RequestHandlerFiltered(Microsoft.AspNetCore.Http.HttpContext httpContext)
+                {
+                    var result = await filteredInvocation(new DefaultEndpointFilterInvocationContext(httpContext));
+                    await GeneratedRouteBuilderExtensionsCore.ExecuteObjectResult(result, httpContext);
+                }
+
+                RequestDelegate targetDelegate = filteredInvocation is null ? RequestHandler : RequestHandlerFiltered;
+                var metadata = inferredMetadataResult?.EndpointMetadata ?? ReadOnlyCollection<object>.Empty;
+                return new RequestDelegateResult(targetDelegate, metadata);
+            }),
+
+        };
+
+        %GENERATEDCODEATTRIBUTE%
```

Comment:
Similar comment as above. Do we need this attribute on these methods when the whole class is marked as generated?

Reply:
I noodled on our use of the `GeneratedCode` attribute here a little bit more. Since the primary goal of the attribute is to make it easier to inspect if a type is codegenned from metadata, I think it makes sense to only include it in types that could be referenced from user code. Specifically, this would be:

- The `SourceKey` definition
- The strongly-typed `Map` overloads

I refactored the code to apply it to these places. Note, I opted to remove it from the static class containing the strongly-typed `Map` overloads since it's unlikely that user code ends up resolving that directly.

Let me know if you think this philosophy makes sense

Reply:
I believe we want the `GeneratedCodeAttribute` on all the classes we generate in this file.

The reason is so things like analyzers, code-style, and code coverage tools don't warn/report things from inside this code. The devs can't change this code, so they shouldn't see tools talking about.

Even for the `internal static class GenerateRouteBuilderEndpoints` class. The Regex generator puts it on the partial methods, and not the class, because the class is partial to the user's code. So it would be incorrect to put it on the whole class, since the user writes code in this class too.

Reply:
> The Regex generator puts it on the partial methods, and not the class, because the class is partial to the user's code. So it would be incorrect to put it on the whole class, since the user writes code in this class too.

This insights makes sense to me.

Here's the modification I made:

- Moved the `GenerateRouteBuilderEndpoints` to the code generated in the `Microsoft.AspNetCore.Builder` namespace to fix the `global::` using issue
- Added the `GeneratedCode` attribute to the `GenerateRouteBuilderEndpoints` class

So, all the classes we codegen inside that namespacce are marked.

Classes marked as `file private` are not marked, which I assume is fine.

Reply:
> Classes marked as file private are not marked, which I assume is fine.

I think they should be marked as well. That way they don't show up in code coverage reports, that devs can't do anything about.